client/X11/xf_cliprdr.c

```diff
@@ -1663,6 +1928,334 @@ static UINT xf_cliprdr_clipboard_file_range_failure(wClipboardDelegate* delegate
 	return clipboard->context->ClientFileContentsResponse(clipboard->context, &response);
 }
 
+static void xf_cliprdr_fuse_getattr(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info* fi)
+{
+	double timeout = 0;
+	struct stat stbuf;
+
+	xfCliprdrFuseInode* node;
+	xfClipboard* clipboard = (xfClipboard*)fuse_req_userdata(req);
+
+	ArrayList_Lock(clipboard->ino_list);
+	node = (xfCliprdrFuseInode*)ArrayList_GetItem(clipboard->ino_list, ino - 1);
+	ArrayList_Unlock(clipboard->ino_list);
```

Comment:
I'm not sure it's okay to release the lock right after the item has been found. It can be removed from the list and freed while we're keeping a pointer to it here, if the user is really anxious and keeps slamming Ctrl-C. Ditto for other accesses to `clipboard->ino_list`.

Reply:
i'll assign all needed  pointer's member to local var  between the lock to avoid missing pointer.